sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala

```diff
@@ -266,6 +264,12 @@ class ContinuousExecution(
         SQLExecution.withNewExecutionId(
           sparkSessionForQuery, lastExecution)(lastExecution.toRdd)
       }
+    } catch {
+      case t: Throwable
+          if StreamExecution.isInterruptionException(t) && state.get() == RECONFIGURING =>
+        stopSources()
+        sparkSession.sparkContext.cancelJobGroup(runId.toString)
```

Comment:
nit: I think its cleaner to put this in the `finally` since this is the invariant we want when this entire method terminates.